sdk/javascript/src/utils/arrayHelpers.js

```diff
@@ -0,0 +1,130 @@
+const { BufferView } = require('./BufferView');
+
+const readArrayImpl = (bufferInput, FactoryClass, accessor, shouldContinue) => {
+	const view = new BufferView(bufferInput);
+	const elements = [];
+	let previousElement = null;
+	let i = 0;
+	while (shouldContinue(i, view)) {
+		const element = FactoryClass.deserialize(view.buffer);
+
+		if (0 >= element.size)
+			throw RangeError('element size has invalid size');
+
+		if (accessor && previousElement && accessor(previousElement) >= accessor(element))
+			throw RangeError('elements in array are not sorted');
+
+		elements.push(element);
+		view.shiftRight(element.size);
+
+		previousElement = element;
+		++i;
+	}
+
+	return elements;
+};
+
+const writeArrayImpl = (output, elements, count, accessor = null) => {
+	for (let i = 0; i < count; ++i) {
+		const element = elements[i];
+		if (accessor && 0 < i && accessor(elements[i - 1]) >= accessor(element))
+			throw RangeError('array passed to writeArray is not sorted');
+
+		output.write(element.serialize());
+	}
+};
+
+const arrayHelpers = {
+	/**
+	 * Calculates aligned size.
+	 * @param {number} size Size.
+	 * @param {number} alignment Alignment.
+	 * @returns {number} Size alingned up to alignment.
+	 */
+	alignUp: (size, alignment) => Math.floor((size + alignment - 1) / alignment) * alignment,
+
+	/**
+	 * Reads array of objects.
+	 * @param {Uint8Array} bufferInput A uint8 array.
+	 * @param {type} FactoryClass Factory used to deserialize objects.
+	 * @param {Function} accessor Optional accessor used to check objects order.
```

Comment:
order => ordering
Function => function

Reply:
I thought that type is actually Function not function?

Reply:
we're using function everywhere ^^